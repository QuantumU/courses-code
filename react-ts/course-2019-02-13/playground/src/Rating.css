.star-active {
  color: gold;
}

.star-inactive {
  color: lightgrey;
}