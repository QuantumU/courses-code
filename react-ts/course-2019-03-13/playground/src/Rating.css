.star {
  font-size: 3rem;
  color: lightgrey;
  user-select: none;
}

.star.active {
  color: gold;
}
