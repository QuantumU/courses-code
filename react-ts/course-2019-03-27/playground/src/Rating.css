.star-inactive {
  color: lightgrey;
  font-size: 40px;
}

.star-active {
  color: gold;
  font-size: 40px;
}