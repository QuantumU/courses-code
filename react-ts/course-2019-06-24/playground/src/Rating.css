.star {
  font-size: 30px;
  color: lightgrey;
  user-select: none;
}

.star.active {
  color: gold;
}
