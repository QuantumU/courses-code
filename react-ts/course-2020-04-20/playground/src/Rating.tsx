import React, { FC } from "react";
import styled from "styled-components";

type StarProps = {
  active: boolean;
};

const Star = styled.span`
  color: ${(props: StarProps) => (props.active ? "gold" : "lightgrey")};
`;

const RatingContainer = styled.div`
  border: 4px solid black;
`;

type RatingProps = {
  stars: number;
  onStarsChange: (newRating: number) => void;
};

const Rating: FC<RatingProps> = (props) => {
  const starElements = [];
  for (let i = 1; i <= props.stars; i++) {
    starElements.push(
      <Star
        active={true}
        onClick={() => {
          props.onStarsChange(i);
        }}
      >
        ★
      </Star>
    );
  }
  for (let i = props.stars + 1; i <= 5; i++) {
    starElements.push(
      <Star
        active={false}
        onClick={() => {
          props.onStarsChange(i);
        }}
      >
        ☆
      </Star>
    );
  }
  return <RatingContainer>{starElements}</RatingContainer>;
}

export default Rating;
