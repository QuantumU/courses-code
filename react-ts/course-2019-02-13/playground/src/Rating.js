import React, { Component } from 'react';
import './Rating.css';

export default class Rating extends Component {
  /*
    props:
      stars: number
    events:
      onStarsChange: function(number)
  */
  render() {
    const starData = [];
    for (let i = 1; i <= 5; i++) {
      starData.push({
        id: i,
        active: i <= this.props.stars
      });
    }
    return (
      <div>
        {starData.map(star => (
          <span
            key={star.id}
            className={star.active ? 'star-active' : 'star-inactive'}
            onClick={() => {this.props.onStarsChange(star.id)}}
          >
            {star.active ? '★' : '☆'}
          </span>
        ))}
      </div>
    );
  }
}
