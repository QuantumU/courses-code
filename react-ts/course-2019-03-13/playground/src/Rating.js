import React, { Component } from 'react';

import './Rating.css';

class Rating extends Component {
  // props:
  //   stars: number
  // events:
  //   onStarsChange: (stars: number) => void

  shouldComponentUpdate(nextProps, nextState) {
    return this.props.stars !== nextProps.stars;
  }

  render() {
    const starIndexes = [1, 2, 3, 4, 5];

    return (
      <div>
        {starIndexes.map(index =>
          this.props.stars < index ? (
            <span
              className="star"
              key={index}
              onClick={() => {
                this.props.onStarsChange(index);
              }}
            >
              ☆
            </span>
          ) : (
            <span
              className="star active"
              key={index}
              onClick={() => {
                this.props.onStarsChange(index);
              }}
            >
              ★
            </span>
          )
        )}
      </div>
    );
  }
}

export default Rating;
