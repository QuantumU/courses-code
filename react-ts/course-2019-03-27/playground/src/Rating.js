import React from 'react';
import './Rating.css';
import {range} from 'lodash'

const Rating = React.memo(props => {
  /**
   * props:
   *   stars: number
   * event:
   *   onStarsChange: (newStars: number) => undefined
   */
  const starIndexes = range(1, 6);

  return starIndexes.map(index => (
    <span
      className={index <= props.stars ? 'star-active' : 'star-inactive'}
      onClick={() => {
        props.onStarsChange(index);
      }}
      key={index}
    >
      {index <= props.stars ? '★' : '☆'}
    </span>
  ));
});

export default Rating;
