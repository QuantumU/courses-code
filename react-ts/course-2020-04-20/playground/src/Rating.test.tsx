import React from "react";
import { render, fireEvent } from "@testing-library/react";

import Rating from "./Rating";

test("renders three active stars", () => {
  const instance = render(<Rating stars={3} onStarsChange={() => {}} />);
  const fullStars = instance.getAllByText("★");
  expect(fullStars.length).toEqual(3);
});

test("reacts to click on the fourth star", () => {
  const mockFn = jest.fn();
  const instance = render(<Rating stars={3} onStarsChange={mockFn} />);
  const firstEmptyStar = instance.getAllByText("☆")[0];
  fireEvent.click(firstEmptyStar);
  expect(mockFn).toHaveBeenCalledWith(4);
});

it("matches the snapshot", () => {
  const instance = render(<Rating stars={3} onStarsChange={() => {}} />);
  expect(instance.baseElement).toMatchSnapshot();
});
