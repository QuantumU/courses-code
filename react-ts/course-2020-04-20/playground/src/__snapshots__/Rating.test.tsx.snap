// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-AxirZ dCqXhF"
    >
      <span
        class="sc-AxjAm cdvcqO"
      >
        ★
      </span>
      <span
        class="sc-AxjAm cdvcqO"
      >
        ★
      </span>
      <span
        class="sc-AxjAm cdvcqO"
      >
        ★
      </span>
      <span
        class="sc-AxjAm dPfmEN"
      >
        ☆
      </span>
      <span
        class="sc-AxjAm dPfmEN"
      >
        ☆
      </span>
    </div>
  </div>
</body>
`;
